### DISCUSSION TITLE
nfs-utils fails to build

### DISCUSSION THREAD

**Discussion by Photon89** [Freetz-NG/freetz-ng#937] [Allgemein]
Hi guys, I'm new to freetz-ng as I just bought a Fritz Box recently, though I have some experience as Linux user. I am trying to build freetz-ng on Manjaro Linux.

It seems like I am stuck with a variant of https://github.com/Freetz-NG/freetz-ng/issues/410 . Here is what I get when trying to build freetz-ng: https://pastebin.com/Hj0Ru3k2

As you can see from the last few lines, `/usr/lib/libtirpc.so` is present on the file system and part of the package `libtirpc` (version 1.3.4). I couldn't figure out from the comments in #410 what exactly the problem and the workaround is. Basically, in https://github.com/Freetz-NG/freetz-ng/issues/410#issuecomment-974612290 we need to override the system version of the file with the one brought by freetz-ng. But why isn't it configured this way in the first place?

Thanks for any hint!

P.S.: Sorry, hab nicht geschnallt, dass hier größtenteils auf Deutsch geschrieben wird, ihr könnt gerne auf Deutsch antworten!

**Comment by aus-sua** [comment]
> on Manjaro Linux

Das könnte Dein Problem sein ---> [prerequisites ---> Getestete Distributionen](https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES/README.md)

Auch bleiben dann immer noch "die üblichen Verdächtigen":
- install prerequisites
- nicht (sauber) aktualisiertes linux
- immer wieder mitgeschleppte (uralte...) .config
- verkonfiguriertes freetz-NG
- verkonfigurierte vm/vm-parameter/Linux
usw.usw...

Hilfreich/nötig ist auch immer Deine .config und Deine .config.compressed.

Warum schreibe ich das alles (und ggf. auch immer wieder...)?
Nun,
bei einem schnellen Test für eine 7590 mit nfs-utils bekomme ich keine Fehlermeldungen, 
das Image wird sauber erstellt. 
[fmake_+_conf_+_conf.comp_.zip](https://github.com/Freetz-NG/freetz-ng/files/14146037/fmake_%2B_conf_%2B_conf.comp_.zip)
:-)

**Comment by fda77** [comment]
> /usr/lib/libtirpc.so: file not recognized: file format not recognized

Beim crosscompilieren sollten keine Libs des Hostsystems verwendet werden, es kann sein dass man irgend wo den Pfad zu libtirpc setzen muss

> bei einem schnellen Test für eine 7590 mit nfs-utils

Ich weiss zwar nicht von welchem Gerät die Logs sind, es wird aber "__arm__-linux-uclibcgnueabi-gcc" genutzt , und die 7590 ist mips

**Comment by Photon89** [comment]
> Beim crosscompilieren sollten keine Libs des Hostsystems verwendet werden, es kann sein dass man irgend wo den Pfad zu libtirpc setzen muss

Genau, deshalb wundert es mich, dass der Build-Prozess so gebaut ist, dass auf dem Hostsystem nach Libs gesucht wird.

> Ich weiss zwar nicht von welchem Gerät die Logs sind, es wird aber "arm-linux-uclibcgnueabi-gcc" genutzt , und die 7590 ist mips

Sorry, bin nicht auf die Idee gekommen, dass das wichtig sein könnte! Ist eine 7520 bei mir.

> Das könnte Dein Problem sein ---> [prerequisites ---> Getestete Distributionen](https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES/README.md)

Betrachtet mich gerne als Beta-Tester für Arch/Manjaro. ;)

> install prerequisites

rpcgen ist ja im Hostsystem vorhanden, das Problem ist wohl eher, dass es umgekehrt nicht benutzt werden sollte, sondern stattdessen das von freetz-ng mitgelieferte.

> nicht (sauber) aktualisiertes linux
> immer wieder mitgeschleppte (uralte...) .config
> verkonfiguriertes freetz-NG
> verkonfigurierte vm/vm-parameter/Linux

Trifft alles nicht zu. Ist ein sauber aktualisiertes Manjaro Linux (aktuell werden keine neuen Updates angeboten). .config ist frisch, hab gestern erstmals von freetz-ng erfahren. Es ist keine VM sondern eine "bare metal" Installation von Manjaro.

**Comment by fda77** [comment]
Blöd, bei mir hat es für das Gerät auch gebaut

**Reply by aus-sua** [reply]
Bei mir auch schon...
Ubuntu 22.04.3 LTS (Kernel 5.15.0-91-generic x86-64)
[fmake_7520_+_conf_+_conf.comp_.zip](https://github.com/Freetz-NG/freetz-ng/files/14151499/fmake_7520_%2B_conf_%2B_conf.comp_.zip)

> Trifft alles nicht zu. Ist ein sauber aktualisiertes Manjaro Linux

Ist wohl doch das Problem, 
wie wohl immer... [prerequisites nicht beachtet](https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES/README.md)
und/oder "übliche Verdächtige".
Die .config fehlt ja auch...
Schade um die Zeit...

**Comment by Photon89** [comment]
.config ist angehängt! Sorry für den seltsamen Dateinamen, als ".config" wollte GitHub sie nicht haben.
[config.txt](https://github.com/Freetz-NG/freetz-ng/files/14151527/config.txt)

**Comment by aus-sua** [comment]
Wie nun nicht anders zu erwarten,
Deine .config läuft in meiner VM ohne Probleme durch...
[fmake_7520-to_+_conf_+_conf.comp_.zip](https://github.com/Freetz-NG/freetz-ng/files/14151699/fmake_7520-to_%2B_conf_%2B_conf.comp_.zip)

Vorschlag:
 --> [prerequisites beachten](https://github.com/Freetz-NG/freetz-ng/blob/master/docs/PREREQUISITES/README.md)
und dann geht's eben :-)

**Comment by Photon89** [comment]
Glaub ich gerne! Ich würde gerne herausfinden, weshalb es unter Manjaro/Arch nicht geht. Weiß nicht, ob es im Sinne des Erfinders (der Entwickler) ist. :smiley:

**Comment by fda77** [comment]
@aus-sua danke fürs testen

@Photon89 da das Problem wohl nur bei dir auftritt muss du wohl selbst herausfinden weshalb das so ist
